libdnf/dnf-repo.c

```diff
@@ -1215,7 +1339,13 @@ dnf_repo_check_internal(DnfRepo *repo,
         if (!dnf_repo_set_timestamp_modified(repo, error))
             return FALSE;
         age_of_data =(g_get_real_time() - priv->timestamp_modified) / G_USEC_PER_SEC;
-        if (age_of_data > permissible_cache_age) {
+        metadata_expire = dnf_repo_get_metadata_expire(repo);
+        if(metadata_expire != 0)
+            valid_time_allowed = metadata_expire <= permissible_cache_age ? metadata_expire  : permissible_cache_age;
```

Comment:
umm, I am thinking that it might be better if we keep it that way, but I might be wrong.

The part I am worried for removing permissible cache age is that, permissible cache age(I think) is more for the DnfContext( for all repos) each time, while the metadata_expire attribute was valid and can be different for each repo. 

I am going to look more deeply into this, hopefully I can have a more reasonable reply later in the day :D.